backend/plugin/db/mysql/dump.go

```diff
@@ -626,8 +629,8 @@ func getRoutines(txn *sql.Tx, dbName string) ([]*routineSchema, error) {
 	var routines []*routineSchema
 	for _, routineType := range []string{"FUNCTION", "PROCEDURE"} {
 		if err := func() error {
-			query := fmt.Sprintf("SHOW %s STATUS WHERE Db = ?;", routineType)
-			rows, err := txn.Query(query, dbName)
+			query := fmt.Sprintf("SHOW %s STATUS FROM %s;", routineType, dbName)
```

Comment:
Looks like this may not work with MySQL?

Reply:
> Looks like this may not work with MySQL?

I checked the manual again and you are right. For the reason of https://github.com/oceanbase/oceanbase/issues/1335, I will add a type judgment here after my vacation.